src/tools/perf/lib/ucp_tests.cc

```diff
@@ -466,25 +466,54 @@ class ucp_perf_test_runner {
     {
         volatile uint8_t *ptr = (uint8_t*)buffer;
 
-        if ((CMD == UCX_PERF_CMD_PUT) &&
-            (TYPE == UCX_PERF_TEST_TYPE_STREAM_UNI)) {
+        if (use_psn()) {
             while (*ptr != UCP_PERF_LAST_ITER_SN) {
                 progress_responder();
             }
         }
     }
 
+    static void nop_cb(void *request, ucs_status_t status)
+    {
+        ucp_request_free(request);
+    }
+
     /* send the special flag as a last iteration in case of
      * unidirectional PUT test, since responder is waiting for
      * this message */
-    ucs_status_t send_last_iter(ucp_ep_h ep, void *buffer,
+    ucs_status_t send_last_iter(ucp_ep_h ep, void *buffer, size_t size,
                                 uint64_t remote_addr, ucp_rkey_h rkey)
     {
-        if ((CMD == UCX_PERF_CMD_PUT) &&
-            (TYPE == UCX_PERF_TEST_TYPE_STREAM_UNI)) {
+        ucs_status_ptr_t status_p;
+
+        if (use_psn()) {
             fence();
             *(uint8_t*)buffer = UCP_PERF_LAST_ITER_SN;
-            return ucp_put(ep, buffer, sizeof(uint8_t), remote_addr, rkey);
+            /* coverity[switch_selector_expr_is_constant] */
+            switch (CMD) {
+            case UCX_PERF_CMD_PUT:
+                return ucp_put(ep, buffer, sizeof(uint8_t), remote_addr, rkey);
```

Comment:
seems return value is unused
maybe this func should wait for request completion?

Reply:
it does this operation and than flush().
shouldn't flush wait for requests to complete?

Reply:
yes but also need to release the request

Reply:
do it from nop_cb, no?

Reply:
ok